Add MetOffice forecast data and daily update mode

MrHarcombe opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Having converted the UK MetOffice weather integration to use config_flow, I would now like to add forecast data and the daily update mode (and add options_flow to allow the user to change between the two modes in the UI).

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: It builds on from [34900](https://github.com/home-assistant/core/pull/34900) and so possibly supercedes it?
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [X] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [X] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [X] 🥇 Gold - I think?
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

MrHarcombe committed:
Converted MetOffice to use UI for configuration
- Sensor and Weather platforms now use config_flow with unique_ids
- MetOfficeData object is now async-protected
- Integration uses DataUpdateCoordinator and so refreshes correctly
- Test cases written to test config_flow and entity behaviours

MrHarcombe committed:
Converted MetOffice to use UI for configuration
- Sensor and Weather platforms now use config_flow with unique_ids
- MetOfficeData object is now async-protected
- Integration uses DataUpdateCoordinator and so refreshes correctly
- Test cases written to test config_flow and entity behaviours

MrHarcombe committed:
Added further test cases to improve patch coverage. Now tests more negative cases, as well as the positive cases.

MrHarcombe committed:
Added further test cases to improve patch coverage. Now tests more negative cases, as well as the positive cases.

MrHarcombe committed:
- Code changes following reviewer comments on the PR
- Not all sensor entities are now enabled by default
- Changes to the test cases to only therefore test sensor entities that *are* enabled

MrHarcombe committed:
- Code changes following reviewer comments on the PR
- Not all sensor entities are now enabled by default
- Changes to the test cases to only therefore test sensor entities that *are* enabled

MrHarcombe committed:
Now testing metoffice integration entities, so removed from .coveragerc

MrHarcombe committed:
Added forecast and update mode

- Added full forecast data into the weather entity
- Can now switch between "3 hourly" and "daily" updates in the UI

MrHarcombe committed:
Added forecast and update mode

- Added full forecast data into the weather entity
- Can now switch between "3 hourly" and "daily" updates in the UI

MrHarcombe committed:
- Having added forecast data and also daily update modes, also added test cases for these
- Refactored testing as files were getting unmanageably large

MrHarcombe committed:
- Having added forecast data and also daily update modes, also added test cases for these
- Refactored testing as files were getting unmanageably large

MrHarcombe committed:
Further testing coverage

Added test case to test the options_flow UI code
Trapped little bug discovered when switching from hourly to daily in sensors that no longer have a value

MrHarcombe committed:
Further testing coverage

Added test case to test the options_flow UI code
Trapped little bug discovered when switching from hourly to daily in sensors that no longer have a value

MrHarcombe committed:
Didn't make use of options_flow changes

Changes made in options_flow weren't being checked for on restart.
Bumped datapoint API to 0.9.6, loses annoying system.out messages on daily forecast retrievals

MrHarcombe committed:
Better understanding of using ConfigEntry options

From looking at other code examples, now using a common method to determine a value either from options, then data (then a default) in both async_setup_entry and async_show_form.

MrHarcombe committed:
Better understanding of using ConfigEntry options

From looking at other code examples, now using a common method to determine a value either from options, then data (then a default) in both async_setup_entry and async_show_form.

MrHarcombe committed:
Mea culpa

Didn't spot the pylint warning in the workspace until too late, and so it failed in the build. Sorry :(

MrHarcombe committed:
Mea culpa

Didn't spot the pylint warning in the workspace until too late, and so it failed in the build. Sorry :(

teharris1 committed:
Bump pyinsteon to 1.0.3 (#36398)

knyar committed:
Add a Prometheus metric for HVAC action (#31945)

fphammerle committed:
Add huawei_lte sensor for number of unread sms (#35665)

vangorra committed:
Prefer use cloud url for oauth2 for Withings (#36348)

szinn committed:
Allow synology_dsm configuration to specify a host name (#36305)

* Allow configuration to specify a host name.

This will default to "synology", so the sensors would be named
sensor.synology_... which is the original implementation.

By specifying a name, which is required for multiple synology hosts,
you can then have sensor.hostA_... and sensor.hostB_...

* Get the base name for the sensors from the Synology DSM hostname

* Don't need the name for the config

* Added missing import

* Fixed another typo

* Removed definition of unused BASE_NAME

* Removed end of line to revert

* Moved for ordering

bieniu committed:
Add config flow to Dune HD (#36345)

* Add config_flow to the dunehd integration

* Add tests

* Run gen_requirements_all

* Fix pylint error

* Better hostname validation

* Build device info in the class

bramkragten committed:
Update frontend to 20200603.0 (#36405)

celestinjr committed:
Enable handling of 'num_repeats' for itach (#36362)

frenck committed:
Fix dynamically add/remove WLED strip segments (#36407)

bieniu committed:
Fix using the async api from sync context (#36408)

uvjustin committed:
Swap title and album name for streams in forked_daapd (#36381)

hunterjm committed:
update renew logic to work better with cameras responding with invalid termination times by extending the duration (#36393)

Emilv2 committed:
Add new is_reatime attribute for De Lijn (#36369)

jjlawren committed:
Remove deprecated Plex YAML config (#36388)

gladhorn committed:
Pass config into NAD constructor (#34961)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

austinmroczek committed:
Notify user if arming or disarming totalconnect alarm fails (#36085)

gladhorn committed:
Update NAD states only when the device is on (#34809)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

alengwenus committed:
Cover group considers opening and closing states (#36203)

nicx committed:
Add CalDAV upcoming appointments period option (#34584)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bramkragten committed:
Update frontend to 20200603.1 (#36409)

matlimatli committed:
Add support for showing text on Keba EV chargers (#36056)

* Add support for showing text on Keba EV chargers

* Changed implementation to use the notify interface

* Removed stale references to set_text

* Clean up

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

basnijholt committed:
Set automation last_triggered earlier (#35671)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Extend logbook api to be able to fetch a slice of data between two time boundaries. (#35847)

* Create logbook_timeslice api

* add end_datetime

* Add tests

* switch to end_time to match history api

frenck committed:
Bumped version to 0.111.0b0

scop committed:
Upgrade mypy to 0.780 (#36416)

http://mypy-lang.blogspot.com/2020/06/mypy-0780-released.html

frenck committed:
Bumped version to 0.112.0dev0 (#36417)

Quentame committed:
Bump python-synology to 0.8.2 (#36410)

* Bump python-synology to 0.8.2

* state_attributes to device_state_attributes

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

homeassistant committed:
[ci skip] Translation update

CoMPaTech committed:
Cleanup and improve Plugwise merges (#36406)

* Remove period from logging messages

* Save indentation using guard clauses

* Typo

* Walk other files

* Rewalk all files

* Not cleanup, but adding indicatd missing measurements

* Revert new sensors

ctalkington committed:
Update sonarr to 0.2.2 (#36429)

ctalkington committed:
Update sonarr to 0.2.2 (#36429)

balloob committed:
Add partial mobile app sensor validation (#36433)

balloob committed:
Add partial mobile app sensor validation (#36433)

balloob committed:
Add partial mobile app sensor validation (#36433)

Kane610 committed:
Fix deCONZ groups don't report ctmax/min (#36432)

* Groups don't report ctmax/min

Kane610 committed:
Fix deCONZ groups don't report ctmax/min (#36432)

* Groups don't report ctmax/min

balloob committed:
Bump lokalise2 2.5.1 (#36430)

CoMPaTech committed:
Add additional sensors to Plugwise integration (#36431)

frenck committed:
Upgrade numpy to 1.18.5 (#36434)

Kane610 committed:
Remove Axis option to enable/disable camera (#36420)

ziv1234 committed:
Fix Dynalite to explicitly check valid device class (#36418)

* changed back to check for class in DEVICE_CLASSES

* created a flow that would go through everything as it was blocking the commit
and the cv rules prevent an input that would get to that flow

* moved DEFAULT_COVER_CLASS from const to cover

balloob committed:
Guard blowing up converting 0 mired/kelvin (#35486)

balloob committed:
Guard blowing up converting 0 mired/kelvin (#35486)

bachya committed:
Add config flow for Tile (#36173)

* Overhaul Tile

* Adjust coverage

* Fix tests

* Code review

* Code review

* Remove unused config flow step

* Revert "Remove unused config flow step"

This reverts commit cb206e044672deb7f681d2a3ae0be03762854fc0.

* Fix tests

bdraco committed:
Ensure verbose logging flag is respected. (#36444)

bdraco committed:
Ensure verbose logging flag is respected. (#36444)

ctalkington committed:
Add roku exception handling for service calls (#36328)

ctalkington committed:
Add roku exception handling for service calls (#36328)

balloob committed:
Fix invalid device info for Daikin devices (#36448)

balloob committed:
Fix invalid device info for Daikin devices (#36448)

balloob committed:
Bumped version to 0.111.0b1

emontnemery committed:
Bump pychromecast to 6.0.0 (#36414)

* Revert "Prevent race in pychromecast.start_discovery (#36350)"

This reverts commit 391983a0cf56a226120057390ddd33586019b827.

* Adapt to pychromecast 6.0.0

emontnemery committed:
Bump pychromecast to 6.0.0 (#36414)

* Revert "Prevent race in pychromecast.start_discovery (#36350)"

This reverts commit 391983a0cf56a226120057390ddd33586019b827.

* Adapt to pychromecast 6.0.0

shbatm committed:
Fix error on empty UOM for ISY994 Climate Device (#36454)

shbatm committed:
Fix error on empty UOM for ISY994 Climate Device (#36454)

homeassistant committed:
[ci skip] Translation update

lindsaymarkward committed:
Fix yeelight_sunflower hs_color using RGB values (#36470)

lindsaymarkward committed:
Fix yeelight_sunflower hs_color using RGB values (#36470)

balloob committed:
Bump hass-nabucasa to 0.34.5 (#36461)

balloob committed:
Bump hass-nabucasa to 0.34.5 (#36461)

balloob committed:
Use builtin mock (#36473)

balloob committed:
Use builtin mock (#36473)

MrHarcombe commented:
As far as I'm aware this is ready to go? It's in review - am I needing to
do anything else with it?

On Fri, 5 Jun 2020, 12:56 stale[bot], <notifications@github.com> wrote:

> There hasn't been any activity on this pull request recently. This pull
> request has been automatically marked as stale because of that and will be
> closed if no further activity occurs within 7 days.
> Thank you for your contributions.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/core/pull/35195#issuecomment-639436165>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQU6AY4WX4BA66JPEWISQTRVDMPJANCNFSM4MY432TA>
> .
>

bdraco committed:
Update myq for latest api changes (#36469)

bdraco committed:
Update myq for latest api changes (#36469)

bdraco committed:
Update myq for latest api changes (#36469)

bdraco committed:
Upgrade pysonos to 0.0.31 (#36483)

bdraco committed:
Upgrade pysonos to 0.0.31 (#36483)

balloob committed:
panel_custom: Allow ES5+latest build, deprecate html (#36464)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

bachya committed:
Clean up Tile refactor (#36450)

* Clean up Tile refactor

* Code review

papajojo committed:
Add notify_events notify integration (#36049)

* Add new notify_events notification component

* Fix manifest

* Formatting fix

* Black formatting + CODEOWNERS

* Fix requirements_all.txt

* Flake8 warning fix

* Isort fixes

* Update notify-events library requirement version

* Replace Exception to more suitable and update lib version

* Reformat integration according to "0007-integration-config-yaml-structure.md"

* Update homeassistant/components/notify_events/manifest.json

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Fix manifest + remove async

* Black formatting

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

knyar committed:
Add @knyar as a codeowner for prometheus integration (#36487)

frenck committed:
Fix iOS app crashing on None values in Zeroconf service info (#36490)

frenck committed:
Fix iOS app crashing on None values in Zeroconf service info (#36490)

frenck committed:
Fix iOS app crashing on None values in Zeroconf service info (#36490)

balloob committed:
Bumped version to 0.110.5

jrester committed:
Update tesla-powerwall to 0.2.10 (#36486)

modified:   homeassistant/components/powerwall/manifest.json
	modified:   requirements_all.txt
	modified:   requirements_test_all.txt

jrester committed:
Update tesla-powerwall to 0.2.10 (#36486)

modified:   homeassistant/components/powerwall/manifest.json
	modified:   requirements_all.txt
	modified:   requirements_test_all.txt

balloob committed:
Merge pull request #36492 from home-assistant/0.110.5

Adminiuga committed:
Process events from ZHA Window Covering Remote (#36489)

Adminiuga committed:
Process events from ZHA Window Covering Remote (#36489)

bdraco committed:
Upgrade zeroconf to 0.27.1 (#36277)

bdraco committed:
Upgrade zeroconf to 0.27.1 (#36277)

balloob committed:
Bumped version to 0.111.0b2

balloob committed:
Merge remote-tracking branch 'origin/master' into rc

homeassistant committed:
[ci skip] Translation update

thomasloven committed:
Add entity list to light and cover group attributes (#36477)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>
Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bramkragten committed:
Update frontend to 20200603.2 (#36494)

bramkragten committed:
Update frontend to 20200603.2 (#36494)

matgad committed:
Bump version zigpy-cc (#36506)

matgad committed:
Bump version zigpy-cc (#36506)

timvancann committed:
Add Avri config flow (#34288)

* Add config flow to Avri integration

* Add config flow validation

* Update .coveragerc

* Start adding config flow tests

* Fix failing test

* Fix pylint

* Update homeassistant/components/avri/config_flow.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Update homeassistant/components/avri/config_flow.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Fix import order

* Code review comments

* Update homeassistant/components/avri/sensor.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Remove device information

Co-authored-by: J. Nick Koston <nick@koston.org>

Commit:
Add devolo binary_sensor devices (#36370)

* initial commit

* small corrections

* fix linting error

* add new files to coveragerc

* rename devolo_sensor to devolo_device

* use correct import

* use binary_switch platform

* use binary_switch platform

* add binary_sensor to coverage

* adjustments according PR review

* make super call easier to read

* use f-string instead of concatenating

* update docstrings - remove device_id property

* add will_remove_from_hass

balloob committed:
Update netdisco (#36499)

balloob committed:
Update netdisco (#36499)

scop committed:
Add and fix type hints (#36501)

* Fix exceptions.Unauthorized.permission type

* Use auth.permission consts more

* Auth typing improvements

* Helpers typing improvements

* Calculate self.state only once

BKPepe committed:
Upgrade youtube_dl to version 2020.06.06 (#36493)

elupus committed:
Arcam config flow (#34384)

Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

homeassistant committed:
[ci skip] Translation update

definitio committed:
Set state for MQTT entities to 'unavailable' when no connection to broker (#36479)

* Report 'unavailable' state when not connected

to MQTT broker

* Fix tests

* Rewrite to remove the polling

* Add tests

* Add some fixes

JeffLIrion committed:
Fix usage of states 'idle' and 'standby' for Android TV (#36509)

emontnemery committed:
Remove unused parameter for MQTT discovery (#36526)

* Remove unused parameter for MQTT discovery

* Fix tests

dingusdk committed:
Update ihcsdk to 2.7.0 (#36527)

markusressel committed:
Upgrade XS1 component to xs1-api-client v3.0.0 (#36500)

frenck committed:
Fix WLED power and brightness with WLED 0.10+ (#36529)

frenck committed:
Fix WLED power and brightness with WLED 0.10+ (#36529)

homeassistant committed:
[ci skip] Translation update

balloob committed:
Bumped version to 0.111.0b3

Hedda committed:
Update ZHA config flow Zigbee radio description (#35607)

* Update ZHA config flow Zigbee radio description
Update ZHA config flow Zigbee radio descriptions to match docs https://github.com/home-assistant/home-assistant.io/pull/13437

* Make protocol plus model names more readable for end-users
* Update homeassistant/components/zha/core/const.py
Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

* Update homeassistant/components/zha/core/const.py
Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

* Update homeassistant/components/zha/core/const.py
Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

* Zigbee radio description simplified and examples  listed in alphabetical order

shbatm committed:
Add Z-Wave Notification Sensor support to ISY994 (#36548)

shbatm committed:
Add Z-Wave Notification Sensor support to ISY994 (#36548)

Mic92 committed:
Fix intent component initialisation (#36064)

The intent component expect this method from every module that is called intent.
Fixes #35522

Mic92 committed:
Fix intent component initialisation (#36064)

The intent component expect this method from every module that is called intent.
Fixes #35522

prystupa committed:
Add plum_lightpad unique ids (#36549)

Commit:
Improve devolo Home Control (#36557)

Commit:
Change logging of successful scans to debug priority (#36538)

Adminiuga committed:
Prevent double ZHA channel initialization (#36554)

* Preven double channel initialization.

* Use a setter for setting ZHA device availability.

jthure committed:
Add Xiaomi miio vaccum goto service (#35737)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

z00nx committed:
Updated tesla-powerwall to 0.2.11 (#36568)

mreiling committed:
Add services to bypass and unbypass zones on NX584 (#36401)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

gieljnssns committed:
Sonos queue (#35817)

* Add playbase as a model

* Sonos queue

* Match counting of the queue with service play_queue

* Add service remove_from_queue

* Change description of remove_from_queue

* Use ATTR_QUEUE_POSITION instead of ATTR_CURRENT_QUEUE_POSITION

* Removed queue and show queue_position only when playing local

* Add property queue_position

* Return None instead of -1

* Change docstring from -1 to None

ehendrix23 committed:
Update aioharmony to include support for XMPP connectivity to Hub (#36485)

* Update aioharmony to 0.2.3

* Update aioharmony to 0.2.4

frenck committed:
Fix mobile_app registering/update sensor values with an unknown state (#36566)

frenck committed:
Fix mobile_app registering/update sensor values with an unknown state (#36566)

balloob committed:
Bumped version to 0.110.6

balloob committed:
Mobile app fixes (#36559)

balloob committed:
Mobile app fixes (#36559)

stevusprimus committed:
Add service select scene to Yamaha Hifi media player (#36564)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

balloob committed:
Merge pull request #36579 from home-assistant/110.6

balloob committed:
Merge remote-tracking branch 'origin/master' into rc

frenck committed:
Fix mobile_app sensor re-registration handling (#36567)

frenck committed:
Fix mobile_app sensor re-registration handling (#36567)

balloob committed:
Use TestCase.addCleanup (#36560)

balloob committed:
Bumped version to 0.111.0b4

basnijholt committed:
bump aiokef to 0.2.10 (#36574)

0.2.9 generated a lot of calls on the event loop.

basnijholt committed:
bump aiokef to 0.2.10 (#36574)

0.2.9 generated a lot of calls on the event loop.

ianperrin committed:
Add status_light to Sonos set_option service (#35515)

homeassistant committed:
[ci skip] Translation update

jhenkens committed:
Remove EU code from honeywell integration (#36456)

balloob committed:
Fix default for loading games file ps4 (#36592)

balloob committed:
Fix default for loading games file ps4 (#36592)

felipediel committed:
Clean up command learning in the Broadlink integration (#36318)

mreiling committed:
Add NX584 alarm binary_sensor zone_number attribute (#36552)

MartinHjelmare commented:
I'll convert this to draft since this is dependent on another PR:
https://github.com/home-assistant/core/pull/34900

frenck committed:
Upgrade pre-commit to 2.5.0 (#36596)

frenck committed:
Fix mobile_app missing state in sensor registration (#36604)

frenck committed:
Fix mobile_app missing state in sensor registration (#36604)

balloob committed:
Bumped version to 0.110.7

donkawechico committed:
Fix nanoleaf incorrect effect update (#36517)

donkawechico committed:
Fix nanoleaf incorrect effect update (#36517)

mreiling committed:
Change nx584 SCAN_INTERVAL to 10 seconds (#36581)

balloob committed:
Merge pull request #36607 from home-assistant/110.7

balloob committed:
Merge remote-tracking branch 'origin/master' into rc

balloob committed:
Bumped version to 0.111.0b5

bramkragten committed:
Escape <> in owntracks translations (#36612)

bramkragten committed:
Escape <> in owntracks translations (#36612)

balloob committed:
Bump hass-nabucasa to 0.34.6 (#36613)

balloob committed:
Bump hass-nabucasa to 0.34.6 (#36613)

homeassistant committed:
[ci skip] Translation update

MartinHjelmare committed:
Bump coronavirus to 1.1.1 (#36614)

MartinHjelmare committed:
Bump coronavirus to 1.1.1 (#36614)

squishykid committed:
Bump solax library to 0.2.3 (#36600)

* Bump solax library to 0.2.3

* oops

scop committed:
Use past tense in messages for already invalidated deprecated configs (#36591)

balloob committed:
Mark config dependency of frontend (#36587)

bdraco committed:
Fix missing space in powerwall notification message (#36603)

Commit:
Bump dependency & add devices for HomematicIP Cloud (#36595)

* update dependency and test data

* Add test for HmIP-SWDO-PL

* Add device HmIP-MOD-HO

* Fix test

frenck committed:
Bumped version to 0.111.0

fredrike committed:
Fix Daikin config flow for zeroconf devices (#36571)

grogi committed:
Add delay in emulated_hue after PUT (#35307)

frenck committed:
Upgrade pre-commit to 2.5.1 (#36623)

frenck committed:
Merge pull request #36626 from home-assistant/rc

frenck committed:
Merge branch 'master' into dev

shenxn committed:
Fix command line sensors removing quotes with template (#35559)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

engrbm87 committed:
Add Speedtestdotnet config_flow (#36254)

fredrik-rambris committed:
Add ue_smart_radio multiple devices support (#36575)

frenck committed:
Fix name of devolo_home_control integration (#36627)

fronzbot committed:
Add options flow to Blink (#35645)

springstan committed:
Use LENGTH_FEET constant (#34053)

emontnemery committed:
Abort other config flows on import (#36608)

* Abort other flows on import

* Add test

frenck committed:
Remove internal qa scale from numato integration (#36629)

frenck committed:
Use async_on_remove for WLED DataUpdateCoordinator (#36640)

janitha committed:
Allow specifying port for wake_on_lan (#36510)

kangyue92 committed:
Update Baidu TTS to support more voice (#36532)

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Update ZHA dependencies (#36646)

phan-t committed:
Add icon_type as configuration variable (#36594)

starkillerOG committed:
Xiaomi Miio gateway: add ip to zeroconf discovery title (#36653)

* add ip to discovery title of xiaomi miio gateway

* add flow title to identify multiple gateways by IP

* clearify gateway token

* black formatting

* grammer improvements

Co-authored-by: Franck Nijhof <git@frenck.dev>

* grammer improvements

Co-authored-by: Franck Nijhof <git@frenck.dev>

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update pymyq for latest api requirements (#36666)

balloob committed:
Update translations

bdraco committed:
Remove powerwall attributes no longer present in latest firmware (#36667)

frenck committed:
Fix packages when config schema is fully deprecated (#36674)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

Adminiuga committed:
Refactor ZHA Entity availability tracking (#36645)

* Refactor ZHA entity availability

ZHA entity availability tracks on underlying ZHA device availability.

* Update device status without signal.

* Update tests.

* Fix tests.

* Tests for restored devices availability.

* Guard against empty last_seen entry

Refactor device loading a bit.

jjlawren committed:
Fix missing options in Plex config entry (#36683)

zewelor committed:
Fix nightlight support detection (#36656)

homeassistant committed:
[ci skip] Translation update

Adminiuga committed:
Fix ZHA Metering channel formatting method (#36697)

Since zigpy change to support bitmap classes, formatting string was incorrectly generated for the newly joined devices with SmartEnergy metering clusters.

MarBra committed:
Add timers to xiaomi_miio vacuum (#35417)

* Add timers to xiaomi_miio vacuum

* Add timezone

* Add cron and next_schedule

* Use next_schedule from backend library

* Use as_utc from utils

balloob committed:
Improve cloud error handling (#36670)

Adminiuga committed:
Cleanup zha.storage from stale devices (#36700)

Adminiuga committed:
Drop ZHA sensor for Analog/Multistate input clusters (#36696)

zewelor committed:
Remove Yeelight model to device type mapping (#36658)

fredrike committed:
Fix Daikin discovery (#36704)

balloob committed:
Enforce unique ID for Hassio discovery (#36671)

balloob committed:
Guard OwnTracks writing state before hass available (#36680)

frenck committed:
Fix logbook filtering for described events (#36727)

mdegat01 committed:
Add influx 2.0 and InfluxCloud support to InfluxDB integration (#35392)

mreiling committed:
Bump pynx584 to 0.5 (#36615)

mvn23 committed:
Add opentherm_gw.set_hot_water_setpoint service (#34579)

frenck committed:
Migrate script to use describe_event for logbook (#36729)

homeassistant committed:
[ci skip] Translation update

ShaneQi committed:
Implement device_info for LutronCasetaDevice (#36706)

frenck committed:
Upgrade pytest-cov to 2.10.0 (#36720)

balloob committed:
Platforms cleanup when adding entity fails (#36742)

emontnemery committed:
Update config_entry_flow test test_user_has_confirmation (#36633)

* Update test_user_has_confirmation

* Rewrite test according to review comment

bdraco committed:
Fix history timezone with postgres and other dbs (#36746)

Kane610 committed:
Make Axis non-privileged users work again (#36731)

bramkragten committed:
Update frontend to 20200603.3 (#36751)

vangorra committed:
Poll all status data in Vera (#35703)

* Vera now polls for all status data, no only incremental.
Vera polling is not handled using hass event loops with proper backoffs.

* Using long polling.

* Addressing PR feedback.

* Addressing PR feedback.
Adding controller stop on config unload.

bramkragten committed:
Updated frontend to 20200613.0 (#36758)

crazyfx1 committed:
Add sensor platform for vicare integration (heatpump) (#34385)

* Add sensor platform for vicare (heatpump)

* Formatting and fixes

* Formatting and fixes 2

* Fixes and formatting 3

* Fixes and formatting 4

* Add binary_sensor and more sensors

This moves some more climate attributes to sensors and adds
binary_sensors

* Move ActiveError back to climate component

The data returned by ActiveError is more complex.
It takes further investigation on how to interpret it a s a binary sensor.
Therefore it is moved back as an attribute of the climate component

* Update PyViCare library

* PR changes

* PR changes 2

Co-authored-by: Hans Oischinger <hans.oischinger@gmail.com>

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Fix history graphs with mysql/mariadb (#36769)

Adminiuga committed:
Add pi_heating/cooling_demand state attributes (#36775)

frenck committed:
Upgrade wled 0.4.3 (#36760)

prystupa committed:
Refactor plum_lightpad (#36761)

benoitlouy committed:
Set default pjlink timeout  (#36781)

* make pjlink timeout configurable

* fix format

* remove timeout configuration and use a default value instead

balloob committed:
Fix translations download (#36770)

dmulcahey committed:
Bump ZHA quirks (#36797)

scop committed:
Add unique id to huawei_lte config entries (#36765)

vigonotion committed:
Add HVV integration (Hamburg public transportation) (#31564)

Co-authored-by: springstan <46536646+springstan@users.noreply.github.com>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

homeassistant committed:
[ci skip] Translation update

ajschmidt8 committed:
Add alarmdecoder brand specific arming sequences (#36692)

* add adext

* rm new config parameter

* update adext

MrHarcombe committed:
Convert MetOffice to use UI for configuration (#34900)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

MrHarcombe committed:
- Merge metoffice config_flow PR #38900 into metoffice_forecast
- Tidy up forecast code/tests in line with feedback from review of that PR

MrHarcombe commented:
Closing, as the merge from upstream hasn't worked neatly - I'll re-open with a clean PR now I've got a clean merge from PR [#34900](https://github.com/home-assistant/core/pull/34900) with all the review fixes in place.
